File: cpp/scriptedmain.cpp
#include <iostream>
#include "./scriptedmain.h"
using std::cout;
using std::endl;

int meaning_of_life() {
  return 42;
}

#ifdef SCRIPTEDMAIN

int main() {
  cout << "Main: The meaning of life is " << meaning_of_life() << endl;
  return 0;
}

#endif

File: swift/ScriptedMain.swift
import Foundation

public class ScriptedMain {
  public class func meaningOfLife() -> Int {
    return 42
  }

  public init() {}

  public class func main() {
    var meaning = ScriptedMain.meaningOfLife()

    println("Main: The meaning of life is \(meaning)")
  }
}

#if SCRIPTEDMAIN
@objc class ScriptedMainAutoload {
  @objc class func load() {
    ScriptedMain.main()
  }
}
#endif
